website/i18n/zh/docusaurus-plugin-content-blog/2022/03/05/apisix-integration-eureka-service-discovery.md

```diff
@@ -0,0 +1,266 @@
+        weight: 100                  # default weight for node
+        timeout:
+          connect: 2000              # default 2000ms
+          send: 2000                 # default 2000ms
+          read: 5000                 # default 5000ms
+
+```
+
+### 创建路由
+
+创建一个 Route，并在 Upstream 中配置启用 Eureka Service Discovery 插件。
+
+- `upstream.discovery_type` 为 `eureka`。
+- `upstream.service_name` 是应用在 Eureka 中注册的应用名 `A-BOOTIFUL-CLIENT`。
+
+```Shell
+curl http://172.30.45.72:9180/apisix/admin/routes/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
+    "uri": "/*",
+    "host": "eureka-demo",
+    "upstream": {
+        "service_name": "A-BOOTIFUL-CLIENT",
+        "type": "roundrobin",
+        "discovery_type": "eureka"
+    }
+}'
+```
+
+### 请求路由
+
+使用 `curl` 命令进行多次请求，验证负载均衡效果。
+
+```Shell
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8081%
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8080%
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8082%
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8081%
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8080%
+$ curl http://172.30.45.72:9080/ -H "Host: eureka-demo"
+server-8082%
+```
+
+从上述返回结果可以看到，请求被依次分配到 Eureka 中注册的三个实例上，这是因为我们使用的负载均衡算法是 `roundrobin`，所有的后端实例会被轮流分配请求。
+
+### 模拟实例宕机
+
+关闭其中 `8081` 实例，模拟实例宕机场景，观察请求效果。
+
+```Shell
+$ while true; do curl http://172.30.45.72:9080/ -H "Host: eureka-demo"; echo; sleep 1; done
+
+server-8080
+server-8082
+server-8081
+
+server-8080
+server-8082
+server-8081
+
+server-8080
+server-8082
+
+server-8080
+server-8082
+```
+
+由上述结果可以看出，关闭 `8081` 实例后，Apache APISIX 会及时同步到 Eureka 的最新实例列表，然后将请求转发给正确的后端。
+
+### 诊断工具
+
+在微服务系统中，经常会遇到非预期转发的问题，这些问题的原因可能来自服务发现中的各个链路，例如：客户端注册异常，注册中心本身数据异常，网关读取注册数据异常等等，因此发生异常时链路中可以使用的诊断工具将会尤为重要。
+
+所以 APISIX 在 Service Discovery 插件中提供了一个诊断接口，可以方便的查询出当前网关正在使用的服务列表，结合注册中心的控制台，我们就对网关到注册中心这一链路做出快速诊断。
+
+诊断接口默认暴露在回环接口的 `9090` 端口，访问方式为 `GET /v1/discovery/{discovery_type}/dump`，例：
+
+```Shell
+curl http://localhost:9090/v1/discovery/eureka/dump
+
+{
+  "services": {
+    "A-BOOTIFUL-CLIENT": [
+      {
+        "weight": 100,
+        "metadata": {
+          "management.port": "8081"
+        },
+        "host": "192.168.50.164",
+        "port": 8081
+      },
+      {
+        "weight": 100,
+        "metadata": {
+          "management.port": "8080"
+        },
+        "host": "192.168.50.164",
+        "port": 8080
+      },
+      {
+        "weight": 100,
+        "metadata": {
+          "management.port": "8082"
+        },
+        "host": "192.168.50.164",
+        "port": 8082
+      }
+    ]
+  },
+  "config": {
+    "prefix": "\/eureka\/",
+    "timeout": {
+      "connect": 2000,
+      "send": 2000,
+      "read": 5000
+    },
+    "fetch_interval": 30,
+    "host": [
+      "http:\/\/172.23.79.129:8761"
+    ],
+    "weight": 100
+  }
+}
+```
+
+通过上述示例就可以查询到 APISIX 正在使用的 Eureka 数据。
+
+### 总结
```

Comment:
```suggestion
## 总结
```

Reply:
I am pretty sure this should be `##` section topic. Double check all section topics.